eZ/Publish/API/Repository/URLAliasService.php

```diff
@@ -60,29 +65,43 @@ public function createUrlAlias(Location $location, $path, $languageCode, $forwar
      *
      * @return \eZ\Publish\API\Repository\Values\Content\URLAlias
      */
-    public function createGlobalUrlAlias($resource, $path, $languageCode, $forwarding = false, $alwaysAvailable = false);
+    public function createGlobalUrlAlias(
+        string $resource,
+        string $path,
+        string $languageCode,
+        bool $forwarding = false,
+        bool $alwaysAvailable = false
+    ): URLAlias;
 
     /**
      * List of url aliases pointing to $location, sorted by language priority.
      *
      * @param \eZ\Publish\API\Repository\Values\Content\Location $location
      * @param bool $custom if true the user generated aliases are listed otherwise the autogenerated
-     * @param string $languageCode filters those which are valid for the given language
+     * @param string|null $languageCode filters those which are valid for the given language
+     * @param bool|null $showAllTranslations If enabled will include all alias as if they where always available.
+     * @param string[]|null $prioritizedLanguageList If set used as prioritized language codes, returning first match.
      *
      * @return \eZ\Publish\API\Repository\Values\Content\URLAlias[]
      */
-    public function listLocationAliases(Location $location, $custom = true, $languageCode = null);
+    public function listLocationAliases(
+        Location $location,
+        bool $custom = true,
+        ?string $languageCode = null,
+        bool $showAllTranslations = null,
+        array $prioritizedLanguageList = null
```

Comment:
If this is intended to be an array then why it is nullable? Can't you use an empty array as a default value?

Reply:
> Can't you use an empty array as a default value?

Unfortunately no. `null` is an indicator for SA layer that `$showAllTranslations` and `$prioritizedLanguageList` value should be resolved base on current site access settings.  An empty array means "all languages" (ref. `\eZ\Publish\API\Repository\Values\Content\Language::ALL`). Any other values are used as-is. 

It's not the best API design but PR is not intended to change the behavior of any of refactored methods.